From 766a8d202a9de38175034f47cf3118785cd9d166 Mon Sep 17 00:00:00 2001
From: tangchen <tangchen@cn.fujitsu.com>
Date: Mon, 6 Jan 2014 16:43:54 +0800
Subject: [PATCH] ACPI / tables: Check if id is NULL in acpi_table_parse()

References: Memory node hot-remove (fate#316483)
Patch-mainline: v3.14
Git-commit: de2d1a7e9310c4d1464faf469c737f08b5608600

strncmp() does not check if the params are NULL. In acpi_table_parse(),
if @id is NULL, the kernel will panic.

Signed-off-by: Tang Chen <tangchen@cn.fujitsu.com>
Acked-by: Toshi Kani <toshi.kani@hp.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 drivers/acpi/tables.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/acpi/tables.c b/drivers/acpi/tables.c
index 380eedd..8bc807a 100644
--- a/drivers/acpi/tables.c
+++ b/drivers/acpi/tables.c
@@ -294,7 +294,7 @@ int __init acpi_table_parse(char *id, acpi_tbl_table_handler handler)
 	if (acpi_disabled)
 		return -ENODEV;
 
-	if (!handler)
+	if (!id || !handler)
 		return -EINVAL;
 
 	if (strncmp(id, ACPI_SIG_MADT, 4) == 0)
